pulsar-client/src/main/java/org/apache/pulsar/client/impl/ResetCursorData.java

```diff
@@ -43,6 +43,21 @@ public ResetCursorData(long ledgerId, long entryId, boolean isExcluded) {
         this.isExcluded = isExcluded;
     }
 
+    // Private constructor used only for json deserialization
+    private ResetCursorData(String position) {
+        if ("latest".equals(position)) {
+            this.ledgerId = Long.MAX_VALUE;
+            this.entryId = Long.MAX_VALUE;
+        } else if ("earliest".equals(position)) {
+            this.ledgerId = -1;
+            this.entryId = -1;
+        } else {
+            throw new IllegalArgumentException(
```

Comment:
It looks like we are not supporting the same encoding of MessageIdImpl and you are only supporting the two constant values.

Am I missing something?

Reply:
Just like the MessageIdImpl, the encoding format like `{"ledgerId":9223372036854775807,"entryId":9223372036854775807,"partitionIndex":-1}
` is not implemented through this constructor, but directly through deserialization of the json object. This constructor is only for the case of constant values.

Reply:
@eolivelli PTAL.